dotnet/src/Connectors/Connectors.Memory.DuckDB/Database.cs

```diff
@@ -121,19 +109,20 @@ private static string EncodeFloatArrayToString(float[]? data)
     public async IAsyncEnumerable<DatabaseEntry> GetNearestMatchesAsync(
         DuckDBConnection conn,
         string collectionName,
-        float[]? embedding,
+        float[] embedding,
         int limit,
         double minRelevanceScore = 0,
         [EnumeratorCancellation] CancellationToken cancellationToken = default)
     {
-        var embeddingArrayString = EncodeFloatArrayToString(embedding ?? Array.Empty<float>());
+        var embeddingArrayString = EncodeFloatArrayToString(embedding);
 
         using var cmd = conn.CreateCommand();
         cmd.CommandText = $@"
-            SELECT key, metadata, timestamp, cast(embedding as string) as embeddingAsString, cast(cosine_similarity(embedding,{embeddingArrayString}) as FLOAT) as score FROM {TableName}
-            WHERE collection=?1 AND score >= {minRelevanceScore.ToString("F12", CultureInfo.InvariantCulture)}
+            SELECT key, metadata, timestamp, embedding, (embedding <=> {embeddingArrayString}) as score FROM {TableName}
+            WHERE collection=?1 AND len(embedding) > 0 AND score >= {minRelevanceScore.ToString("F12", CultureInfo.InvariantCulture)}
```

Comment:
DuckDB and DuckDB.NET now support named parameters so you can give parameters names for better readability.

Reply:
there is no support for parameters that are array of floats

Reply:
Right, you can't pass an array as a parameter but I meant `?1`. You can replace it with a better name if you want to.

Reply:
what is the syntax for that? `?parameterName`?

Reply:
`$parameterName`

Reply:
Done!